src/components/organisms/Topnavigation/Topnavigation.tsx

```diff
@@ -0,0 +1,122 @@
+import "./Topnavigation.css";
+import "../../atoms/buttons/Hamburger";
+import Hamburger from "../../atoms/buttons/Hamburger";
+import { useEffect, useRef, useState } from "react";
+import CartMenu, { CartMenuData } from "../../molecules/CartMenu/CartMenu";
+import Menu, { MenuData } from "../../molecules/Menu/Menu";
+
+export interface TopnavigationProps {
+  title: string;
+  cart: string;
+  cart_content: string;
+  divider: string;
+  cartMenu: CartMenuData;
+  menu: MenuData[];
+}
+
+const TopNavigation = (props: TopnavigationProps) => {
+  const topnavigationProps = props;
+
+  const [isCartMenuOpen, setIsCartMenuOpen] = useState<boolean>(false);
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const [hamburgerStyle, setHamburgerStyle] = useState<string>("bar");
```

Comment:
Create enum for hamburger style and use that

Reply:
Added